city_scrapers/spiders/stl_port_authority.py

```diff
@@ -0,0 +1,203 @@
+
+
+class StlPortAuthority(CityScrapersSpider):
+    name = "stl_port_authority"
+    agency = "St. Louis Port Authority Commission"
+    timezone = "America/Chicago"
+    custom_settings = {"ROBOTSTXT_OBEY": False}
+
+    def __init__(self, *args, **kwargs):
+        self.agenda_map = defaultdict(list)
+        super().__init__(*args, **kwargs)
+
+    def start_requests(self):
+        url = (
+            "https://www.stlouis-mo.gov/government/departments/sldc/boards/"
+            "documents/port-authority-commission-documents.cfm"
+        )
+        yield scrapy.Request(
+            url=url, method="GET", callback=self._get_agenda_urls, priority=100
+        )
+
+        calendar_urls = [
+            (
+                "https://www.stlouis-mo.gov/events/"
+                "past-meetings.cfm?span=-30&department=272"
+            ),
+            "https://www.stlouis-mo.gov/events/all-public-meetings.cfm?span=30",
+        ]
+        for url in calendar_urls:
+            yield scrapy.Request(url=url, method="GET", callback=self.parse)
+
+    def parse(self, response):
+        for url in self._get_event_urls(response):
+            yield scrapy.Request(url, callback=self._parse_event, dont_filter=True)
+
+    def _get_agenda_urls(self, response):
+        links = response.css("a.CS_PgIndex_Item_Title::attr(href)").getall()
+        urls = links[:3]
+        for url in urls:
+            url = response.urljoin(url)
+            yield scrapy.Request(
+                url=url, method="GET", callback=self._parse_links, priority=100
+            )
+
+    def _get_event_urls(self, response):
+        event_urls = response.css("ul.list-group h4 a::attr(href)").getall()
+        event_sponsors = response.css("ul.list-group li span.small::text").getall()
+        urls = []
+        for url, sponsor in zip(event_urls, event_sponsors):
+            if "port authority" in sponsor.lower():
+                urls.append(response.urljoin(url))
+        return urls
+
+    def _parse_event(self, response):
+        """
+        return Meeting items.
+        """
+        start = self._parse_start(response)
+        links_key = datetime.strftime(start, "%m-%d")
+
+        meeting = Meeting(
+            title=self._parse_title(response),
+            description="",
+            classification=BOARD,
+            start=start,
+            end=self._parse_end(response),
+            all_day=False,
+            location=self._parse_location(response),
+            source=response.url,
+        )
+
+        if links_key in self.agenda_map.keys():
+            meeting["links"] = self.agenda_map[links_key]
+        else:
+            meeting["links"] = []
+
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+        return meeting
+
+    def _parse_title(self, response):
+        """Parse or generate meeting title."""
+        title = response.css("div.page-title-row h1::text").get().strip()
+        return title
```

Comment:
The title can just be "Port Authority Commission" since its the same for every meeting.